src/sentry/event_manager.py

```diff
@@ -585,9 +587,32 @@ def save(
 
         self._data = job["event"].data.data
 
+        # Check if the project is configured for auto upgrading and we need to upgrade
+        # to the latest grouping config.
+        if auto_upgrade_grouping and project.get_option("sentry:grouping_auto_update"):
+            _auto_update_grouping(project)
+
         return job["event"]
 
 
+def _auto_update_grouping(project):
+    if not settings.SENTRY_GROUPING_AUTO_UPDATE_ENABLED:
+        return
+
+    old_grouping = project.get_option("sentry:grouping_config")
+    new_grouping = DEFAULT_GROUPING_CONFIG
+
+    # update to latest grouping config but not if a user is already on
+    # beta.
+    if old_grouping != new_grouping and old_grouping != BETA_GROUPING_CONFIG:
```

Comment:
How do orgs leave beta configs automatically?

Reply:
At the moment there is only one beta config and we just want to remove this thing entirely going forward. The future of hierarchical grouping (which is the beta grouping) is a bit unclear at the moment. So we will manually transition them out once we have a path forward and then retire the beta grouping concept.